tests/phpunit/tests/hooks/wpHook/arrayAccess.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ * Test ArrayAccess methods of WP_Hook.
+ *
+ * @group hooks
+ * @covers WP_Hook::offsetGet
+ * @covers WP_Hook::offsetSet
+ * @covers WP_Hook::offsetUnset
+ */
+class Tests_Hooks_WpHook_ArrayAccess extends WP_UnitTestCase {
```

Comment:
It's arguable whether the `offsetGet|Set|Unset()` should be in separate test files, as just because they're interface methods doesn't necessarily mean they behave as they should. With there only being one test method, I get the logic in just putting it in one class though, for now if not forever.